### PR TITLE
Adding SeriesView object that allows inplace operations

### PR DESCRIPTION
**Pull request by devin-petersohn**
* Now `df['column1'].fillna(0, inplace=True)` will work correctly
* Wraps all Series objects that are returned from getitem
* Minor fix to an issue in setitem where some Series objects were
  incorrectly set based on their index

## Related issue number

- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] passes `black --check modin/`

### CODE REVIEW

File: modin/pandas/series.py [hunk 1]
```diff
@@ -15,7 +15,162 @@ def na_op():
...
+                            self.parent_df[self._loc] = self.series
+                        else:
+                            self.parent_df.reindex(index=self.series.index, copy=False)
+                        return None
+                    else:
+                        return self.series.__getattribute__(item)(*args, **kwargs)
+
+                method = inplace_handler
+            return method
+        elif item not in default_behaviors:
```
Comment (simon-mo) [inline modin/pandas/series.py]:
```suggestion
         elif item in default_behaviors:
```
IDK if this is the intention, but we have two repeated `item not in default_behaviors`

File: modin/pandas/series.py [hunk 2]
```diff
@@ -15,7 +15,162 @@ def na_op():
...
+        ]
+        if item not in default_behaviors:
+            method = self.series.__getattribute__(item)
+            if (
+                callable(method)
+                and "inplace" in str(inspect.signature(method))
+                and self.parent_df is not None
+            ):
+
+                def inplace_handler(*args, **kwargs):
```
Comment (simon-mo) [inline modin/pandas/series.py]: This is quite brilliant! Can you document this part?

File: modin/pandas/series.py (hunk 1 above)
Comment (devin-petersohn) [reply modin/pandas/series.py]: Good catch, the second conditional was created during development for testing and should be removed.

File: modin/pandas/series.py [hunk 3]
```diff
@@ -15,7 +15,243 @@ def na_op():
...
+
+    Important: This is not needed to replace every Series in Modin. For example, when an
+        operation on a Series returns a new Series, it does not need to return an object
+        of this class. It can return a Series because the new object does not have a
+        DataFrame that it is associated with.
+
+    """
+
+    def __init__(self, series=None, parent_df=None, loc=None):
+        self.series = series
```
Comment (williamma12) [inline modin/pandas/series.py]: Can you add some asserts to make sure that all of the arguments are not none?

```suggestion
        self.series = series
        assert isinstance(parent_df) is modin.DataFrame
        assert loc != None
```

File: modin/pandas/series.py [hunk 4]
```diff
@@ -15,7 +15,243 @@ def na_op():
...
+                    Returns:
+                        If `inplace` is True: None, else: A new Series.
+                    """
+                    if kwargs.get("inplace", False):
+                        prev_len = len(self.series)
+                        self.series.__getattribute__(item)(*args, **kwargs)
+                        if prev_len == len(self.series):
+                            self.parent_df.loc[self._loc] = self.series
+                        else:
+                            self.parent_df.reindex(index=self.series.index, copy=False)
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/pandas/series.py]: `reindex(copy=False)` updates the `parent_df` inplace.

File: modin/pandas/series.py [hunk 5]
```diff
@@ -15,7 +15,243 @@ def na_op():
...
+        `loc`/`iloc` return a Series as well.
+
+    Important: This is not needed to replace every Series in Modin. For example, when an
+        operation on a Series returns a new Series, it does not need to return an object
+        of this class. It can return a Series because the new object does not have a
+        DataFrame that it is associated with.
+
+    """
+
+    def __init__(self, series, parent_df, loc):
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/pandas/series.py]: Sure!

**Pull request merged**